platform/features/plot/test/services/ExportImageServiceSpec.js

```diff
@@ -0,0 +1,122 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * Open MCT includes source code licensed under additional open source
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+
+/**
+ * ExportImageServiceSpec. Created by hudsonfoo on 09/03/16.
+ */
+define(
+    ["../../src/services/ExportImageService"],
+    function (ExportImageService) {
+        var mockQ,
+            mockDeferred,
+            mockPromise,
+            mockTimeout,
+            mockLog,
+            mockHtml2Canvas,
+            mockJsPDF,
+            mockJsPDFSave,
+            mockSaveAs,
+            mockExportTimeoutConstant,
+            exportImageService;
+
+        describe("ExportImageService", function () {
+            beforeEach(function () {
+                mockDeferred = jasmine.createSpyObj(
+                    "deferred",
+                    ["reject", "resolve"]
+                );
+                mockPromise = jasmine.createSpyObj(
+                    "promise",
+                    ["then", "finally"]
+                );
+                mockPromise.then = function (callback) {
+                    callback();
+                };
+                mockQ = {
+                    "defer": function () {
+                        return {
+                            "resolve": mockDeferred.resolve,
+                            "reject": mockDeferred.reject,
+                            "promise": mockPromise
+                        };
+                    }
+                };
+                mockTimeout = function (fn, time) {
+                    return {
+                        "cancel": function () {}
+                    };
+                };
+                mockLog = jasmine.createSpyObj(
+                    "$log",
+                    ["warn"]
+                );
+                mockHtml2Canvas = jasmine.createSpy("html2canvas");
+                mockJsPDFSave = jasmine.createSpy("jsPDFSave");
+                mockJsPDF = function () {
+                    return {
+                        "addImage": function () {},
+                        "save": mockJsPDFSave
+                    };
+                };
+                mockSaveAs = jasmine.createSpy("saveAs");
+                mockExportTimeoutConstant = 0;
+
+                exportImageService = new ExportImageService(
+                    mockQ,
+                    mockTimeout,
+                    mockLog,
+                    mockExportTimeoutConstant,
+                    mockHtml2Canvas,
+                    mockJsPDF,
+                    mockSaveAs
+                );
+            });
+
```

Comment:
I think the verification of high-level interactions is good here, but would also like to pass in a `testElement` (don't think there's anything that needs to be mocked, so this could just be some object) to the `exportXYZ` methods, in order to `expect(mockHtml2Canvas).toHaveBeenCalledWith(testElement, { onrendered: jasmine.any(Function) })`

Testing of the `onrendered` callback itself would also be good. In particular, it would be nice to check interactions with the canvas object to make sure exportPNG isn't producing JPGs and so forth.